lib/active_merchant/billing/gateways/square_new.rb

```diff
@@ -0,0 +1,297 @@
+        if address = options[addr_key] || options[:address] || options[:billing_address]
+          add_address_for(post, address, addr_key)
+        end
+      end
+
+      def add_address_for(post, address, addr_key)
+        addr_key = addr_key.to_sym
+        post[addr_key] ||= {} # Or-Equals in case they passed in using Square's key format
+        if address[:name]
+          post[addr_key][:first_name] = address[:name].split(" ")[0]
+          post[addr_key][:last_name] = address[:name].split(" ")[1] if address[:name].split(" ")[1]
+        end
+        post[addr_key][:address_line_1] = address[:address1] if address[:address1]
+        post[addr_key][:address_line_2] = address[:address2] if address[:address2]
+        post[addr_key][:address_line_3] = address[:address3] if address[:address3]
+
+        post[addr_key][:locality] = address[:city] if address[:city]
+        post[addr_key][:sublocality] = address[:sublocality] if address[:sublocality]
+        post[addr_key][:sublocality_2] = address[:sublocality_2] if address[:sublocality_2]
+        post[addr_key][:sublocality_3] = address[:sublocality_3] if address[:sublocality_3]
+
+        post[addr_key][:administrative_district_level_1] = address[:state] if address[:state]
+        if address[:administrative_district_level_2] # In the US, this is the county.
+          post[addr_key][:administrative_district_level_2] = address[:administrative_district_level_2]
+        end
+        post[addr_key][:postal_code] = address[:zip] if address[:zip]
+        post[addr_key][:country] = address[:country] if address[:country]
+      end
+
+      def add_idempotency_key(post, options)
+        post[:idempotency_key] = options[:order_id] || generate_unique_id
+      end
+
+      def add_amount(post, money, options)
+        currency = options[:currency] || currency(money)
+        post[:amount_money] = {
+          amount: localized_amount(money, currency).to_i,
+          currency: currency.upcase
+        }
+      end
+
+      def add_descriptor(post, options)
+        return unless options[:descriptor]
+
+        post[:statement_description_identifier] = options[:descriptor]
+      end
+
+      def create_post_for_purchase(money, payment, options)
+        post = {}
+
+        post[:source_id] = payment
+        post[:customer_id] = options[:customer] if options[:customer].present?
+        post[:note] = options[:description] if options[:description]
+        post[:location_id] = @location_id if @location_id
+
+        add_idempotency_key(post, options)
+        add_amount(post, money, options)
+
+        post
+      end
+
+      def sdk_request(api_name, method, body, params = {})
+        parameters = body ? { body: body }.merge(params) : params
+
+        raw_response = square_client.send(api_name).send(method, parameters)
+        log(raw_response)
+
+        parse(raw_response)
+      end
+
+      def commit(api_name, method, body, params = {})
```

Comment:
hmm, correct me if I'm wrong, but it looks like we can change those last two params: body and params into a single one like just params? 
I saw that you called this method in few places passing nil as body, so instead of doing this:
```ruby
commit(:customers, :delete_customer, nil, customer_id: identification)
```
you could just do this:
```ruby
commit(:customers, :delete_customer, customer_id: identification)
```
Same situation in this example:
```
commit(:customers, :update_customer, post, customer_id: identification)
```
you could just add customer id to the post hash
```ruby
post[:customer_id] = identification
commit(:customers, :update_customer, post)
```
What do you think?

Reply:
You're right! Refactored.